pkgs/shells/bash/5.nix

```
@@ -27,19 +27,19 @@ let
       inherit sha256;
     }
   );
+  patch_suffix = "p${toString (builtins.length upstreamPatches)}";
 in
 lib.warnIf (withDocs != null)
   ''
     bash: `.override { withDocs = true; }` is deprecated, the docs are always included.
   ''
   stdenv.mkDerivation
-  (fa: {
+  (finalAttrs: {
     pname = "bash${lib.optionalString interactive "-interactive"}";
-    version = "5.3${fa.patch_suffix}";
-    patch_suffix = "p${toString (builtins.length upstreamPatches)}";
+    version = "5.3${patch_suffix}";
 
     src = fetchurl {
-      url = "mirror://gnu/bash/bash-${lib.removeSuffix fa.patch_suffix fa.version}.tar.gz";
+      url = "mirror://gnu/bash/bash-${lib.removeSuffix patch_suffix finalAttrs.version}.tar.gz";
```

I'm guessing that the stdenv rebuild is coming from here.

```suggestion
      url = "mirror://gnu/bash/bash-${lib.removeSuffix finalAttrs.patch_suffix finalAttrs.version}.tar.gz";
```

---

Right, and your refactoring is changing the expression because it no longer depends on finalAttrs, causing a stdenv rebuild. A refactor like this should cause 0 rebuilds, and I'm thinking it will once you revert this part